### ISSUE TITLE
question about appearance embedding

### ISSUE THREAD

**Issue by Torment123** [city-super/Scaffold-GS#38]
Hi, I see that the appearance embedding is constructed based on the number of views in train cameras, and when shifting to eval mode, the uid of the test camera is directly used to query the learned embedding.

If I understand the appearance embedding correctly, it is set up so that the view-dependent effect can be better encoded; but since the test cameras and train cameras have different views, and so their uids have different meaning in this aspect, I think querying the same learned embedding would lead to wrong effect ? Thanks

**Comment by tongji-rkr** [comment]
In the experiment with the appearance embedding, we do not divide the training set and test set, and the same appearance embedding is used for all views in the rendering process after training.

**Comment by Torment123** [comment]
> In the experiment with the appearance embedding, we do not divide the training set and test set, and the same appearance embedding is used for all views in the rendering process after training.

thanks for the clarification

**Comment by xuxumiao777** [comment]
> In the experiment with the appearance embedding, we do not divide the training set and test set, and the same appearance embedding is used for all views in the rendering process after training.

I tried scaffold gs in imw2020 dataset. The renderings have quite different appearances after training even though they are rendered with the same appearance id. Why?

**Comment by Torment123** [comment]
> > In the experiment with the appearance embedding, we do not divide the training set and test set, and the same appearance embedding is used for all views in the rendering process after training.
> 
> I tried scaffold gs in imw2020 dataset. The renderings have quite different appearances after training even though they are rendered with the same appearance id. Why?

Hi, is the rendering still reasonably good or bad? is there a train / test split in your dataset? If so, since the test views are different from the train views, the same appearance id cannot be used in such cases. Like explained by the author, when appearance ids are used, they do not split the dataset
